### DISCUSSION TITLE
emqx mixed content

### DISCUSSION THREAD

**Discussion by nahuel2** [emqx/emqx#8284] [Q&A]
Dear, I have developed an iot system with emqx and nodejs. The system runs on AWS and I have a certificate for the AWS domain. 
![image](https://user-images.githubusercontent.com/17711623/174665061-84d5c334-fefd-4426-9f06-2a215a9b7772.png)

There are internal processes of my API that are not secure. They are http and not https. One of them is EMQX

![image](https://user-images.githubusercontent.com/17711623/174665318-3e7e2f00-cf03-4358-8ca9-b77674088440.png)

I've been searching the forums and testing without success. The system works fine on mqtt:ws:8083 but not on mqtt:wss:8084
These are my listeners

![image](https://user-images.githubusercontent.com/17711623/174665839-a24923e3-ac33-4f5f-bbfa-67c2067d2bd4.png)

This is my configuration in docker compose

emqx:
    container_name: MQX
    image: emqx/emqx:4.2.14

    ports:
      - 18083:18083 #Dashboard
      - 18084:18084 #Dashboard SSL
      - 1883:1883
      - 8883:8883
      - 8083:8083
      - 8084:8084
      - 8085:8081

    volumes:
      - vol-emqx-data:/opt/emqx/data
      - vol-emqx-log:/opt/emqx/log

    #Los links son porque mongo, emqx estan en diferentes dokers. Con los links los puedo conectar entre si
    links:
      - mongo
      #Extra hosts me sirve para mapear diferentes direcciónes
    extra_hosts:
      - "localnode:xxx.xxx.xxx.xxx"

   
 environment:
      EMQX_NAME: idtsur-emqx
      EMQX_HOST: 127.0.0.1
      TZ: "${TZ}"

      #DEFAULT DASH PASS
      EMQX_DASHBOARD__DEFAULT_USER__PASSWORD: "${EMQX_DEFAULT_USER_PASSWORD}"

      # DEFAULT APP SECRET
      EMQX_MANAGEMENT__DEFAULT_APPLICATION__SECRET: "${EMQX_DEFAULT_APPLICATION_SECRET}"

      #Anonimos no, solo con credenciales
      EMQX_ALLOW_ANONYMOUS: "false"
      EMQX_NOMATCH: "deny"

      # MONGO CONNECTION
      EMQX_AUTH__MONGO__TYPE: single
      EMQX_AUTH__MONGO__TOPOLOGY__POOL_SIZE: 1
      EMQX_AUTH__MONGO__TOPOLOGY__MAX_OVERFLOW: 0

      EMQX_AUTH__MONGO__SERVER: "mongo:${MONGO_EXT_PORT}"
      EMQX_AUTH__MONGO__POOL: 8 #Colecciones vivas en paralelo
      #DEFAULT DASH PASS
      EMQX_DASHBOARD__DEFAULT_USER__PASSWORD: "${EMQX_DEFAULT_USER_PASSWORD}"

      # DEFAULT APP SECRET
      EMQX_MANAGEMENT__DEFAULT_APPLICATION__SECRET: "${EMQX_DEFAULT_APPLICATION_SECRET}"

      #Anonimos no, solo con credenciales
      EMQX_ALLOW_ANONYMOUS: "false"
      EMQX_NOMATCH: "deny"

      # MONGO CONNECTION
      EMQX_AUTH__MONGO__TYPE: single
      EMQX_AUTH__MONGO__TOPOLOGY__POOL_SIZE: 1
      EMQX_AUTH__MONGO__TOPOLOGY__MAX_OVERFLOW: 0

      EMQX_AUTH__MONGO__SERVER: "mongo:${MONGO_EXT_PORT}"
      EMQX_AUTH__MONGO__POOL: 8 #Colecciones vivas en paralelo

      # HABILITAR LISTAS DE CONTROL DE ACCESO (ACL)
      EMQX_AUTH__MONGO__LOGIN: "${MONGO_USERNAME}"
      EMQX_AUTH__MONGO__PASSWORD: "${MONGO_PASSWORD}"
      EMQX_AUTH__MONGO__AUTH_SOURCE: xxxx
      EMQX_AUTH__MONGO__DATABASE: "xxxx"
      #USUARIOS COMUNES
      EMQX_AUTH__MONGO__AUTH_QUERY__COLLECTION: "emqxauthrules"
      EMQX_AUTH__MONGO__AUTH_QUERY__PASSWORD_HASH: plain
      EMQX_AUTH__MONGO__AUTH_QUERY__PASSWORD_FIELD: "password" #campo, puede cambiar por eso pongo el campo donde debe buscar
      EMQX_AUTH__MONGO__AUTH_QUERY__SELECTOR: "username=%u"

      #super usuario no usamos mucho xq se resuelve de otra forma pero esta buena la info para proyectos
      EMQX_AUTH__MONGO__SUPER_QUERY__COLLECTION: "emqxauthrules"
      EMQX_AUTH__MONGO__SUPER_QUERY__SUPER_FIELD: "is_superuser"
      EMQX_AUTH__MONGO__SUPER_QUERY__SELECTOR: "username=%u"
      EMQX_AUTH__MONGO__SUPER_QUERY: "off" #lo dejo apagado

      # HABILITAR LISTAS DE CONTROL DE ACCESO (ACL)

      EMQX_AUTH__MONGO__ACL_QUERY: "on"
      EMQX_AUTH__MONGO__ACL_QUERY__COLLECTION: "emqxauthrules"
      EMQX_AUTH__MONGO__ACL_QUERY__SELECTOR: "username=%u"

      #ARRANQUE DE PLUGINS
      EMQX_LOADED_PLUGIND: "emqx_recon,emqx_retainer,emqx_management,emqx_dashboard,emqx_auth_mongo"

      EMQX_LISTENER__TCP__EXTERNAL__MAX_CONNECTIONS: 1000

my api .env

MQTT_PREFIX=wss://
MQTT_HOST=my elastic IP
MQTT_PORT=8084

ports in aws

![image](https://user-images.githubusercontent.com/17711623/174667845-8ac5e597-8710-42f2-9e61-71b251e945f7.png)

and load balancer configuration

![image](https://user-images.githubusercontent.com/17711623/174668050-f131fa4b-90da-4855-a2cb-d70511f0e4ea.png)

Please, if someone can guide me how to make it work, I will be very grateful.

Best regards

**Comment by zmstone** [comment]
Hi @nahuel2 
The opensource edition EMQX version 4.2 is out of maintenance window since April 2022.
We encourage you to upgrade to EMQX 5.0.0 where listener configuration (including certificates upload) can be done from the dashboard UI.

**Comment by nahuel2** [comment]
Hi @zmstone, thanks for you answer. I will update to version 5, best regards

**Reply by nahuel2** [reply]
Hi @zmstone, upgrade to version 5 but got this error when starting docker-compose

[nahuel@localhost services]$ docker-compose up
Starting EMQX ... done
Attaching to EMQX
EMQX    | rpc.port_discovery = EMQX_RPC__PORT_DISCOVERY = manual
EMQX    | log.file_handlers.default.enable = EMQX_LOG__FILE_HANDLERS__DEFAULT__ENABLE = false
EMQX    | log.console_handler.enable = EMQX_LOG__CONSOLE_HANDLER__ENABLE = true
EMQX    | node.name = EMQX_NODE__NAME = idtsur-emqx@127.0.0.1
EMQX    | Listener ssl:default on :8883 started.
EMQX    | Listener tcp:default on :1883 started.
EMQX    | Listener ws:default on :8083 started.
EMQX    | Listener wss:default on :8084 started.
EMQX    | 2022-06-23T20:16:46.006020+00:00 [error] Generic server emqx_telemetry terminating. Reason: {{badmatch,{aborted,{bad_type,{telemetry,'idtsur-emqx@127.0.0.1',<<"67AE203E-F331-11EC-B59A-6B60CC692170">>}}}},[{emqx_telemetry,ensure_uuids,0,[{file,"emqx_telemetry.erl"},{line,576}]},{emqx_telemetry,handle_continue,2,[{file,"emqx_telemetry.erl"},{line,145}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,695}]},{gen_server,loop,7,[{file,"gen_server.erl"},{line,437}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}. Last message: {continue,init}. State: undefined.
EMQX    | 2022-06-23T20:16:46.006624+00:00 [error] crasher: initial call: emqx_telemetry:init/1, pid: <0.2253.0>, registered_name: emqx_telemetry, error: {{badmatch,{aborted,{bad_type,{telemetry,'idtsur-emqx@127.0.0.1',<<"67AE203E-F331-11EC-B59A-6B60CC692170">>}}}},[{emqx_telemetry,ensure_uuids,0,[{file,"emqx_telemetry.erl"},{line,576}]},{emqx_telemetry,handle_continue,2,[{file,"emqx_telemetry.erl"},{line,145}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,695}]},{gen_server,loop,7,[{file,"gen_server.erl"},{line,437}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}, ancestors: [emqx_modules_sup,<0.2251.0>], message_queue_len: 1, messages: [{'$gen_call',{<0.2251.0>,[alias|#Ref<0.2939178059.2922184708.255674>]},enable}], links: [<0.2252.0>], dictionary: [], trap_exit: false, status: running, heap_size: 4185, stack_size: 29, reductions: 11345; neighbours:
EMQX    | 2022-06-23T20:16:46.006924+00:00 [error] Supervisor: {local,emqx_modules_sup}. Context: child_terminated. Reason: {{badmatch,{aborted,{bad_type,{telemetry,'idtsur-emqx@127.0.0.1',<<"67AE203E-F331-11EC-B59A-6B60CC692170">>}}}},[{emqx_telemetry,ensure_uuids,0,[{file,"emqx_telemetry.erl"},{line,576}]},{emqx_telemetry,handle_continue,2,[{file,"emqx_telemetry.erl"},{line,145}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,695}]},{gen_server,loop,7,[{file,"gen_server.erl"},{line,437}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}. Offender: id=emqx_telemetry,pid=<0.2253.0>.
EMQX    | 2022-06-23T20:16:46.007485+00:00 [error] crasher: initial call: application_master:init/4, pid: <0.2250.0>, registered_name: [], exit: {{bad_return,{{emqx_modules_app,start,[normal,[]]},{'EXIT',{{{badmatch,{aborted,{bad_type,{telemetry,'idtsur-emqx@127.0.0.1',<<"67AE203E-F331-11EC-B59A-6B60CC692170">>}}}},[{emqx_telemetry,ensure_uuids,0,[{file,"emqx_telemetry.erl"},{line,576}]},{emqx_telemetry,handle_continue,2,[{file,"emqx_telemetry.erl"},{line,145}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,695}]},{gen_server,loop,7,[{file,"gen_server.erl"},{line,437}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]},{gen_server,call,[emqx_telemetry,enable,15000]}}}}},[{application_master,init,4,[{file,"application_master.erl"},{line,142}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}, ancestors: [<0.2249.0>], message_queue_len: 1, messages: [{'EXIT',<0.2251.0>,normal}], links: [<0.2249.0>,<0.1680.0>], dictionary: [], trap_exit: true, status: running, heap_size: 610, stack_size: 29, reductions: 197; neighbours:
EMQX    | 2022-06-23T20:16:46.007814+00:00 [error] Generic server emqx_telemetry terminating. Reason: {{badmatch,{aborted,{bad_type,{telemetry,'idtsur-emqx@127.0.0.1',<<"67AE761A-F331-11EC-8A92-B7B46ED4B010">>}}}},[{emqx_telemetry,ensure_uuids,0,[{file,"emqx_telemetry.erl"},{line,576}]},{emqx_telemetry,handle_continue,2,[{file,"emqx_telemetry.erl"},{line,145}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,695}]},{gen_server,loop,7,[{file,"gen_server.erl"},{line,437}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}. Last message: {continue,init}. State: undefined.
EMQX    | 2022-06-23T20:16:46.008480+00:00 [critical] app: emqx_modules, line: 88, mfa: emqx_machine_boot:start_one_app/1, msg: failed_to_start_app, reason: {emqx_modules,{bad_return,{{emqx_modules_app,start,[normal,[]]},{'EXIT',{{{badmatch,{aborted,{bad_type,{telemetry,'idtsur-emqx@127.0.0.1',<<"67AE203E-F331-11EC-B59A-6B60CC692170">>}}}},[{emqx_telemetry,ensure_uuids,0,[{file,"emqx_telemetry.erl"},{line,576}]},{emqx_telemetry,handle_continue,2,[{file,"emqx_telemetry.erl"},{line,145}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,695}]},{gen_server,loop,7,[{file,"gen_server.erl"},{line,437}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]},{gen_server,call,[emqx_telemetry,enable,15000]}}}}}}
EMQX    | 2022-06-23T20:16:46.008729+00:00 [error] Supervisor: {local,emqx_machine_sup}. Context: start_error. Reason: {'EXIT',{{failed_to_start_app,emqx_modules,{emqx_modules,{bad_return,{{emqx_modules_app,start,[normal,[]]},{'EXIT',{{{badmatch,{aborted,{bad_type,{telemetry,'idtsur-emqx@127.0.0.1',<<"67AE203E-F331-11EC-B59A-6B60CC692170">>}}}},[{emqx_telemetry,ensure_uuids,0,[{file,"emqx_telemetry.erl"},{line,576}]},{emqx_telemetry,handle_continue,2,[{file,"emqx_telemetry.erl"},{line,145}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,695}]},{gen_server,loop,7,[{file,"gen_server.erl"},{line,437}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]},{gen_server,call,[emqx_telemetry,enable,15000]}}}}}}},[{emqx_machine_boot,start_one_app,1,[{file,"emqx_machine_boot.erl"},{line,89}]},{lists,foreach,2,[{file,"lists.erl"},{line,1342}]},{emqx_machine_boot,post_boot,0,[{file,"emqx_machine_boot.erl"},{line,38}]},{supervisor,do_start_child_i,3,[{file,"supervisor.erl"},{line,414}]},{supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,400}]},{supervisor,'-start_children/2-fun-0-',3,[{file,"supervisor.erl"},{line,384}]},{supervisor,children_map,4,[{file,"supervisor.erl"},{line,1250}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,350}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,423}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,390}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}}. Offender: id=emqx_machine_boot,pid=undefined.
EMQX    | 2022-06-23T20:16:46.009395+00:00 [error] crasher: initial call: application_master:init/4, pid: <0.1928.0>, registered_name: [], exit: {{{shutdown,{failed_to_start_child,emqx_machine_boot,{'EXIT',{{failed_to_start_app,emqx_modules,{emqx_modules,{bad_return,{{emqx_modules_app,start,[normal,[]]},{'EXIT',{{{badmatch,{aborted,{bad_type,{telemetry,'idtsur-emqx@127.0.0.1',<<"67AE203E-F331-11EC-B59A-6B60CC692170">>}}}},[{emqx_telemetry,ensure_uuids,0,[{file,"emqx_telemetry.erl"},{line,576}]},{emqx_telemetry,handle_continue,2,[{file,"emqx_telemetry.erl"},{line,145}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,695}]},{gen_server,loop,7,[{file,"gen_server.erl"},{line,437}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]},{gen_server,call,[emqx_telemetry,enable,15000]}}}}}}},[{emqx_machine_boot,start_one_app,1,[{file,"emqx_machine_boot.erl"},{line,89}]},{lists,foreach,2,[{file,"lists.erl"},{line,1342}]},{emqx_machine_boot,post_boot,0,[{file,"emqx_machine_boot.erl"},{line,38}]},{supervisor,do_start_child_i,3,[{file,"supervisor.erl"},{line,414}]},{supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,400}]},{supervisor,'-start_children/2-fun-0-',3,[{file,"supervisor.erl"},{line,384}]},{supervisor,children_map,4,[{file,"supervisor.erl"},{line,1250}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,350}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,423}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,390}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}}}},{emqx_machine_app,start,[normal,[]]}},[{application_master,init,4,[{file,"application_master.erl"},{line,142}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}, ancestors: [<0.1927.0>], message_queue_len: 1, messages: [{'EXIT',<0.1929.0>,normal}], links: [<0.1927.0>,<0.1680.0>], dictionary: [], trap_exit: true, status: running, heap_size: 1598, stack_size: 29, reductions: 250; neighbours:
EMQX    | Listener ssl:default on :8883 stopped.
EMQX    | Listener tcp:default on :1883 stopped.
EMQX    | Listener ws:default on :8083 stopped.
EMQX    | Listener wss:default on :8084 stopped.
EMQX    | [os_mon] memory supervisor port (memsup): Erlang has closed
EMQX    | [os_mon] cpu supervisor port (cpu_sup): Erlang has closed
EMQX    | {"Kernel pid terminated",application_controller,"{application_start_failure,emqx_machine,{{shutdown,{failed_to_start_child,emqx_machine_boot,{'EXIT',{{failed_to_start_app,emqx_modules,{emqx_modules,{bad_return,{{emqx_modules_app,start,[normal,[]]},{'EXIT',{{{badmatch,{aborted,{bad_type,{telemetry,'idtsur-emqx@127.0.0.1',<<\"67AE203E-F331-11EC-B59A-6B60CC692170\">>}}}},[{emqx_telemetry,ensure_uuids,0,[{file,\"emqx_telemetry.erl\"},{line,576}]},{emqx_telemetry,handle_continue,2,[{file,\"emqx_telemetry.erl\"},{line,145}]},{gen_server,try_dispatch,4,[{file,\"gen_server.erl\"},{line,695}]},{gen_server,loop,7,[{file,\"gen_server.erl\"},{line,437}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,226}]}]},{gen_server,call,[emqx_telemetry,enable,15000]}}}}}}},[{emqx_machine_boot,start_one_app,1,[{file,\"emqx_machine_boot.erl\"},{line,89}]},{lists,foreach,2,[{file,\"lists.erl\"},{line,1342}]},{emqx_machine_boot,post_boot,0,[{file,\"emqx_machine_boot.erl\"},{line,38}]},{supervisor,do_start_child_i,3,[{file,\"supervisor.erl\"},{line,414}]},{supervisor,do_start_child,2,[{file,\"supervisor.erl\"},{line,400}]},{supervisor,'-start_children/2-fun-0-',3,[{file,\"supervisor.erl\"},{line,384}]},{supervisor,children_map,4,[{file,\"supervisor.erl\"},{line,1250}]},{supervisor,init_children,2,[{file,\"supervisor.erl\"},{line,350}]},{gen_server,init_it,2,[{file,\"gen_server.erl\"},{line,423}]},{gen_server,init_it,6,[{file,\"gen_server.erl\"},{line,390}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,226}]}]}}}},{emqx_machine_app,start,[normal,[]]}}}"}
EMQX    | Kernel pid terminated (application_controller) ({application_start_failure,emqx_machine,{{shutdown,{failed_to_start_child,emqx_machine_boot,{'EXIT',{{failed_to_start_app,emqx_modules,{emqx_modules,{bad_return,{{emqx_modules_app,start,[normal,[]]},{'EXIT',{{{badmatch,{aborted,{bad_type,{telemetry,'idtsur-emqx@127.0.0.1',<<"67AE203E-F331-11EC-B59A-6B60CC692170">>}}}},[{emqx_telemetry,ensure_uuids,0,[{file,"emqx_telemetry.erl"},{line,576}]},{emqx_telemetry,handle_continue,2,[{file,"emqx_telemetry.erl"},{line,145}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,695}]},{gen_server,loop,7,[{file,"gen_server.erl"},{line,437}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]},{gen_server,call,[emqx_telemetry,enable,15000]}}}}}}},[{emqx_machine_boot,start_one_app,1,[{file,"emqx_machine_boot.erl"},{line,89}]},{lists,foreach,2,[{file,"lists.erl"},{line,1342}]},{emqx_machine_boot,post_boot,0,[{file,"emqx_machine_boot.erl"},{line,38}]},{supervisor,do_start_child_i,3,[{file,"supervisor.erl"},{
EMQX    | 
EMQX    | Crash dump is being written to: log/erl_crash.dump...done
EMQX exited with code 1

**Reply by nahuel2** [reply]
my docker-compose v3.8

version: '3.8'

volumes:
  vol-emqx-data:
    name: foo-emqx-data
  #   vol-emqx-etc:
  #     name: foo-emqx-etc
  vol-emqx-log:
    name: foo-emqx-log

services:
  #mongo:
    #container_name: mongo
    #image: mongo:5.0.2
    #restart: always
    #environment:
      #TZ: "${TZ}"
      #MONGO_INITDB_ROOT_USERNAME: "${MONGO_USERNAME}"
      #MONGO_INITDB_ROOT_PASSWORD: "${MONGO_PASSWORD}"
    #volumes:
      #- ./mongodata:/data/db
    #ports:
      #- ${MONGO_EXT_PORT}:27017

  emqx:
    container_name: EMQX
    image: emqx/emqx:5.0.0
    #restart: always
    ports:
      - 18083:18083 #Dashboard
      - 18084:18084 #Dashboard SSL
      - 1883:1883
      - 8883:8883
      - 8083:8083
      - 8085:8081

    volumes:
      - vol-emqx-data:/opt/emqx/data
      #   - vol-emqx-etc:/opt/emqx/etc
      - vol-emqx-log:/opt/emqx/log

    #Los links son porque mongo, emqx estan en diferentes dokers. Con los links los puedo conectar entre si
    #links:
      #- mongo
      #Extra hosts me sirve para mapear diferentes direcciónes
    #extra_hosts:
      #- "localnode:192.168.1.189"

    environment:
      EMQX_NAME: idtsur-emqx
      EMQX_HOST: 127.0.0.1
      #TZ: "${TZ}"

      #DEFAULT DASH PASS
      #EMQX_DASHBOARD__DEFAULT_USER__PASSWORD: "${EMQX_DEFAULT_USER_PASSWORD}"

      # DEFAULT APP SECRET
      #EMQX_MANAGEMENT__DEFAULT_APPLICATION__SECRET: "${EMQX_DEFAULT_APPLICATION_SECRET}"

      #Anonimos no, solo con credenciales
      #EMQX_ALLOW_ANONYMOUS: "false"
      #EMQX_NOMATCH: "deny"

      # MONGO CONNECTION
      #EMQX_AUTH__MONGO__TYPE: single
      #EMQX_AUTH__MONGO__SERVER: "mongo:${MONGO_EXT_PORT}"
      #EMQX_AUTH__MONGO__POOL: 8 #Colecciones vivas en paralelo
      #EMQX_AUTH__MONGO__LOGIN: "${MONGO_USERNAME}"
      #EMQX_AUTH__MONGO__PASSWORD: "${MONGO_PASSWORD}"
      #EMQX_AUTH__MONGO__AUTH_SOURCE: admin
      #EMQX_AUTH__MONGO__DATABASE: "iotgl"
      
      #USUARIOS COMUNES
      #EMQX_AUTH__MONGO__AUTH_QUERY__COLLECTION: "emqxauthrules"
      #EMQX_AUTH__MONGO__AUTH_QUERY__PASSWORD_HASH: plain
      #EMQX_AUTH__MONGO__AUTH_QUERY__PASSWORD_FIELD: "password" #campo, puede cambiar por eso pongo el campo donde debe buscar
      #EMQX_AUTH__MONGO__AUTH_QUERY__SELECTOR: "username=%u"

      #super usuario no usamos mucho xq se resuelve de otra forma pero esta buena la info para proyectos
      #EMQX_AUTH__MONGO__SUPER_QUERY__COLLECTION: "emqxauthrules"
      #EMQX_AUTH__MONGO__SUPER_QUERY__SUPER_FIELD: "is_superuser"
      #EMQX_AUTH__MONGO__SUPER_QUERY__SELECTOR: "username=%u"
      #EMQX_AUTH__MONGO__SUPER_QUERY: "off" #lo dejo apagado

      # HABILITAR LISTAS DE CONTROL DE ACCESO (ACL)

      #EMQX_AUTH__MONGO__ACL_QUERY: "on"
      #EMQX_AUTH__MONGO__ACL_QUERY__COLLECTION: "emqxauthrules"
      #EMQX_AUTH__MONGO__ACL_QUERY__SELECTOR: "username=%u"

      #ARRANQUE DE PLUGINS
      #EMQX_LOADED_PLUGIND: "emqx_recon,emqx_retainer,emqx_management,emqx_dashboard,emqx_auth_mongo"

      #EMQX_LISTENER__TCP__EXTERNAL__MAX_CONNECTIONS: 1000

**Comment by nahuel2** [comment]
in docker hub there are two possibilities of images for emqx?
in the official one there is no version 5 but in the second option there is. what is the difference?

![image](https://user-images.githubusercontent.com/17711623/175394240-a083d4c1-2601-4f2d-a307-e150680f53ff.png)

Thank you!

**Reply by zmstone** [reply]
Hi @nahuel2 

We have recently packaged version 5 and the CI pipeline auto-pushed to emqx/emqx 
but the official image is not automatically synced, it's so far still pending on a manual step.

@Rory-Z could you help to automate it ?

**Comment by joshemq** [comment]
Hi @nahuel2, I hope your issue has been resolved by moving to EMQX 5. If so, I wonder if you would be interested in sharing more details about your project with me. I'm EMQ's marketing director in the US, and I'd appreciate the chance to interview you and maybe share your project with others if you'd be interested. Would it be okay to contact you directly?

**Reply by nahuel2** [reply]
Hi @joshemq, yes of course, best regarts.
